cli.js

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+
+import { readFile } from 'node:fs/promises'
+import { fileURLToPath } from 'node:url'
+import { dirname, join } from 'node:path'
+
+import { formatHelpMessage, peowly } from 'peowly'
+import { availableChoices } from './lib/options.js'
+import { intersection } from './lib/utils.js'
+
+const resolvedDirname = dirname(fileURLToPath(import.meta.url))
+
+const pkg = JSON.parse(await readFile(join(resolvedDirname, 'package.json'), 'utf8'))
+
+/** @type {Record<Lowercase<import('./lib/options.js').NeostandardChoices>, import('peowly').AnyFlag>} */
+const options = {
+  semi: {
+    listGroup: 'Config options',
+    description: 'Enforce rather than ban semi-colons',
+    type: 'boolean',
+  },
+  nots: {
+    listGroup: 'Config options',
+    description: 'Deactivates all TypeScript based linting',
+    type: 'boolean',
+  },
+  tsforjs: {
+    listGroup: 'Config options',
+    description: 'Activates TypeScript based linting for JavaScript files as well',
+    type: 'boolean',
+  },
+  typechecking: {
+    listGroup: 'Config options',
+    description: 'Activates type checking rules ',
+    type: 'boolean',
+  },
+}
+
+const [execPath, filename, ...args] = process.argv
+
+const {
+  flags,
+  remainderArgs,
+} = peowly({
+  args,
+  help: formatHelpMessage('neostandard', {
+    flags: {
+      ...options,
+      '..': {
+        listGroup: 'ESLint options',
+        description: 'All ESLint options are supported',
+      },
+    },
+    usage: '',
+  }),
+  options,
+  name: 'neostandard',
+  pkg,
+  returnRemainderArgs: true,
+})
+
+const unsupportedFlags = intersection(remainderArgs, [
+  '--config',
+  '--init',
+  '--inspect-config',
+  '--no-config-lookup',
+  '--stdin', // TODO: Would be possible to support
+  '-c',
+])
+
+if (unsupportedFlags.length) {
+  console.error('neostandard can not be used with these ESLint flags:', unsupportedFlags.join(', '))
+  process.exit(1)
+}
+
+const config = availableChoices
+  .map(choice => /** @type {Lowercase<choice>} */ (choice.toLowerCase()))
+  .filter(choice => flags[choice])
+
+const configFile = join(resolvedDirname, config.length ? `configs/neostandard-${config.join('-')}.config.js` : 'index.js')
+
+if (remainderArgs.includes('--debug')) {
+  const { default: { enable } } = await import('debug')
```

Comment:
I would advocate that we use a more robust logger with levels over `debug` and `console`. This is entirely a nitpick and should not be considered my trying to block this, but I have found a lot of value out of cli tooling using a robust logger with log levels instead of two loggers and variable formatting. One great example of this is having to migrate to support `--json` output. If we wanted to add that for better tooling integration we would need to replace all the `console.log` and `debug` calls with something anyway. And then it would all likely be a breaking change, when otherwise we could just convert the formatter in the logger from the "raw" format to serializing it with JSON.

Happy to contribute this, I have used both `pino` and [a home grown one I wrote many years ago](https://github.com/wesleytodd/loggerr) more suited to CLI's because it is synchronous (I do think sync logging in tooling is best for debugging). I would prefer either approach over the `debug` package, but again, not intending to try blocking this over it.